### PR TITLE
Passively fixing issues for upcoming axe-core version 4

### LINKED ISSUE #3324: Update axe 4 violations
# Story

## Description

As a developer of terra, I want to have my components support the latest version of axe

## Motivation

So that I am keeping up to date with various accessibility rule changes and fixes to axe's analysis of those dependencies

## Definition of Done

We'll know we're done when:

* The components are updated so that they will not fail axe testing in either the current version or the new version of axe
* The code is merged to main

### PR DESCRIPTION
**Pull request by BenBoersma**
### Summary

Fixing roles in some packages to not fail axe-core version 4 for future WDIO 6 update and updating jest snapshots for those fixes.

Fixes #3324 

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-form-select/src/shared/_OptGroup.jsx [hunk 1]
```diff
@@ -39,7 +39,7 @@ const OptGroup = ({
 }) => {
   const theme = React.useContext(ThemeContext);
   return (
-    <li className={cx('opt-group', { 'is-disabled': disabled }, theme.className)} role="group">
+    <li className={cx('opt-group', { 'is-disabled': disabled }, theme.className)} role="option">
```
Comment (StephenEsser) [inline packages/terra-form-select/src/shared/_OptGroup.jsx:42]: I'm a little suspicious about this change. Can you post the error message that was being thrown?

Comment (BenBoersma) [reply packages/terra-form-select/src/shared/_OptGroup.jsx]: >  "nodes": [
    {
      "all": [],
      "any": [
        {
          "data": [
            "option"
          ],
          "id": "aria-required-children",
          "impact": "critical",
          "message": "Required ARIA child role not present: option",
          "relatedNodes": []
        }
      ],
      "failureSummary": "Fix any of the following:\n  Required ARIA child role not present: option",
      "html": "<ul id=\"terra-select-menu-25\" role=\"listbox\" class=\"_Menu-module__menu___33lDo\" aria-label=\"Menu\" aria-activedescendant=\"terra-select-option-green\" tabindex=\"0\">",
      "impact": "critical",
      "none": [],
      "target": [
        "#terra-select-menu-25"
      ]
    }

Was the original error however I am no longer getting that and am instead getting 

>  "nodes": [
    {
      "all": [],
      "any": [
        {
          "data": [
            "aria-selected=\"false\""
          ],
          "id": "aria-allowed-attr",
          "impact": "critical",
          "message": "ARIA attribute is not allowed: aria-selected=\"false\"",
          "relatedNodes": []
        }
      ],
      "failureSummary": "Fix any of the following:\n  ARIA attribute is not allowed: aria-selected=\"false\"",
      "html": "<li class=\"_OptGroup-module__opt-group___hROs6\" role=\"group\" aria-selected=\"false\">",
      "impact": "critical",
      "none": [],
      "target": [
        "._OptGroup-module__opt-group___hROs6[role=\"group\"]:nth-child(1)"
      ]
    }

I'm going to see if i just remove aria-selected off the OptGroup `<li>` fixes the issue

Comment (BenBoersma) [reply packages/terra-form-select/src/shared/_OptGroup.jsx]: Okay so If I remove `aria-selected="false"` from the `<li>` I get the original top error.

File: packages/terra-dropdown-button/CHANGELOG.md [hunk 2]
```diff
@@ -2,11 +2,14 @@
 
 ## Unreleased
 
+* Fixed
+  * Fixed incorrect role for upcoming axe-core version bump
```
Comment (yuderekyu) [inline packages/terra-dropdown-button/CHANGELOG.md:6]: nitpick  - can we specify which role and component piece we updated, and why?

File: packages/terra-form-select/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * Fixed incorrect role for upcoming axe-core version bump
```
Comment (ShettyAkarsh) [inline packages/terra-form-select/CHANGELOG.md:6]: i think we should also mention about the aria tag changes also here.

**Review by mjhenkes — APPROVED** [review]
The code changes look good to me, but I think we need to check with @scottwilmarth to ensure that the a11y changes we're making here are appropriate.

File: packages/terra-form-select/src/shared/_OptGroup.jsx [hunk 4]
```diff
@@ -39,7 +39,7 @@ const OptGroup = ({
 }) => {
   const theme = React.useContext(ThemeContext);
   return (
-    <li className={cx('opt-group', { 'is-disabled': disabled }, theme.className)} role="group">
+    <li className={cx('opt-group', { 'is-disabled': disabled }, theme.className)} role="option" aria-selected="false">
```
Comment (StephenEsser) [inline packages/terra-form-select/src/shared/_OptGroup.jsx:42]: This is one I'm not too sure about but documentation on the right thing to do here is tricky to find.

**Review by StephenEsser — APPROVED** [review]

File: packages/terra-dropdown-button/CHANGELOG.md (hunk 2 above)
Comment (BenBoersma) [reply packages/terra-dropdown-button/CHANGELOG.md]: Updated here: https://github.com/cerner/terra-core/pull/3323/commits/48e71424fd2542e32a670ae28265982b565f3eac

File: packages/terra-form-select/CHANGELOG.md (hunk 3 above)
Comment (BenBoersma) [reply packages/terra-form-select/CHANGELOG.md]: Updated here

**Comment by BenBoersma** [comment]
With offline conversations with @scottwilmarth, we walked through each of the changes and compared before and after with VoiceOver, JAWS, and IOS VoiceOver. The changes in terra-form-radio, and terra-form-select did not negatively impact accessibility, but the change in terra-dropdown-button did lead to a worse accessibility issue, so I have undone the [role change](https://github.com/cerner/terra-core/pull/3323/commits/94e712e9f6a9731bc6c1a4ebf329cfa8daeb17b3#diff-2d14ee7351a7a98841f3c12329a75a289071df6291565a93956f936e399cdab3R55) causing the issue.

**Review by StephenEsser — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Review by dkasper-was-taken — APPROVED** [review]

**Review by jeremyfuksa — APPROVED** [review]

**Pull request merged**